examples/basic/eks_cluster.tf

```diff
@@ -0,0 +1,145 @@
+data "aws_region" "current" {}
+locals {
+  cluster_name = "example"
+
+  vpc_id     = "vpc-123456"
+  subnet_ids = ["xxxxxxxxx", "xyyyyyyyyy"]
+
+  project_tags = {
+    project_name = local.cluster_name
+    repo         = "github.com/mycompany/myrepo"
+    Terraform    = "true"
+    environment  = "live"
+  }
+
+  company_dns_domain = "mycompany.com"
+  dns_subdomain      = ["project_a", "project_b", "project_c"]
+}
+module "eks_cluster_base" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "14.0.0"
+
+  cluster_name = local.cluster_name
+
+  cluster_version = "1.19"
+  cluster_enabled_log_types = [
+    "api",
+    "audit",
+    "authenticator",
+    "controllerManager",
+    "scheduler"
+  ]
+
+  cluster_log_retention_in_days = 30
+
+  write_kubeconfig   = true
+  config_output_path = "${path.root}/"
+  # Use aws cli for authentication
+  kubeconfig_aws_authenticator_command = "aws"
+  kubeconfig_aws_authenticator_command_args = [
+    "--region",
+    data.aws_region.current.name,
+    "eks",
+    "get-token",
+    "--cluster-name",
+    local.cluster_name,
+  ]
+
+  enable_irsa = true
+  vpc_id      = local.vpc_id
+  subnets     = local.subnet_ids
+
+  tags = local.project_tags
+
+  workers_additional_policies = [
+    "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+  ]
+}
+
+module "eks_cluster_utils" {
+  source = "git::https://github.com/PhilStevenson/tfm_aws_eks.git"
+
+  cluster_name            = local.cluster_name
+  cluster_id              = module.eks_cluster_base.cluster_id
+  oidc_provider_arn       = module.eks_cluster_base.oidc_provider_arn
+  cluster_oidc_issuer_url = module.eks_cluster_base.cluster_oidc_issuer_url
+  kubeconfig_filename     = module.eks_cluster_base.kubeconfig_filename
+  cluster_endpoint        = module.eks_cluster_base.cluster_endpoint
```

Comment:
you could pass in the whole `eks_cluster = module.eks_cluster_base` in one variable if it makes sense, although I have seen issues where it triggers recreations  for no reason, so up to you if you want to try it